hnn_core/viz.py

```diff
@@ -0,0 +1,165 @@
+"""Visualization functions."""
+
+# Authors: Mainak Jas <mainak.jas@telecom-paristech.fr>
+#          Sam Neymotin <samnemo@gmail.com>
+
+import numpy as np
+
+
+def plot_dipole(dpl, ax=None, layer='agg', show=True):
+    """Simple layer-specific plot function.
+
+    Parameters
+    ----------
+    dpl : instance of Dipole
+        The Dipole object.
+    ax : instance of matplotlib figure | None
+        The matplotlib axis
+    layer : str
+        The layer to plot. Can be one of
+        'agg', 'L2', and 'L5'
+    show : bool
+        If True, show the figure
+
+    Returns
+    -------
+    fig : instance of plt.fig
+        The matplotlib figure handle.
+    """
+    import matplotlib.pyplot as plt
+
+    if ax is None:
+        fig, ax = plt.subplots(1, 1)
+    if layer in dpl.dpl.keys():
+        ax.plot(dpl.t, dpl.dpl[layer])
+        ax.set_xlabel('Time (ms)')
+        ax.set_title(layer)
+    if show:
+        plt.show()
+    return ax.get_figure()
+
+
+def plot_hist_input(net, ax=None, show=True):
+    """Plot the histogram of input.
+
+    Parameters
+    ----------
+    ax : instance of matplotlib axis | None
+        An axis object from matplotlib. If None,
+        a new figure is created.
+    show : bool
+        If True, show the figure.
+
+    Returns
+    -------
+    fig : instance of matplotlib Figure
+        The matplotlib figure handle.
+    """
+    import matplotlib.pyplot as plt
+    spikes = np.array(sum(net.spikes.times, []))
+    gids = np.array(sum(net.spikes.gids, []))
+    valid_gids = np.r_[[v for (k, v) in net.gid_dict.items()
+                        if k.startswith('evprox')]]
+    mask_evprox = np.in1d(gids, valid_gids)
+    valid_gids = np.r_[[v for (k, v) in net.gid_dict.items()
+                        if k.startswith('evdist')]]
+    mask_evdist = np.in1d(gids, valid_gids)
+    bins = np.linspace(0, net.params['tstop'], 50)
+
+    if ax is None:
+        fig, ax = plt.subplots(1, 1)
+    ax.hist(spikes[mask_evprox], bins, color='r', label='Proximal')
+    ax.hist(spikes[mask_evdist], bins, color='g', label='Distal')
+    plt.legend()
+    if show:
+        plt.show()
+    return ax.get_figure()
+
+
+def plot_spikes(spikes, ax=None, show=True):
```

Comment:
To complement `plot_spikes_hist()`, this function could also be renamed `plot_spikes_raster()`. Since the vast majority of use-cases for viewing a raster plot are for output spikes, I think it makes sense to leave the hard-coded spike types (i.e., cell_types = `['L5_pyramidal', 'L5_basket', 'L2_pyramidal', 'L2_basket']`) as-is.

Reply:
sounds good, I'll change the name.